File: 1-TwoSum/1-TwoSum.java
// Last updated: 25/03/2026, 15:17:58
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> map= new HashMap<>();

        for(int i=0;i<nums.length;i=i+1){
           int lookingfor=target-nums[i];
           if(map.containsKey(lookingfor)){
            return new int[] {
                i,
                map.get(lookingfor)
            };
           }

           map.put(nums[i],i);
        }

        return new int[]{
            -1-1
        };
        
    }
}

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 10/06/2026, 16:37:32
1class Solution {
2public:
3    vector<int> twoSum(vector<int>& nums, int target) {
4        unordered_map<int,int> mp;
5
6        for(int i=0;i<nums.size();i++){
7            int need=target-nums[i];
8
9            if(mp.count(need)){
10                return {mp[need],i};
11            }
12
13            mp[nums[i]]=i;
14        }
15        return {};
16    }
17};